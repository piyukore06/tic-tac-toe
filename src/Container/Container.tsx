import React from 'react';
import { BoxSize, PlayerType } from '../App';

const Container = ({ hasWon, boxState, onIndexClick }: { hasWon: boolean, boxState: Array<PlayerType | ''>, onIndexClick: (index: number) => void}) => {

  return (
    <div className="container">
    {Array(BoxSize).fill(0).map((_, index) => (
      <button disabled={hasWon} key={index} onClick={() => onIndexClick(index)} className="box">{boxState[index]}</button>
    ))}
    </div>
  );
}

export default Container;
