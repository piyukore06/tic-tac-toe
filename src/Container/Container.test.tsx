import React from 'react';
import renderer from 'react-test-renderer';
import Container from './Container';

test('renders proper hasWon and boxState output', () => {
  const component1 = renderer.create(<Container hasWon={false} boxState={Array(9).fill('')}  onIndexClick={() => {}} />);
  let tree1 = component1.toJSON();
  expect(tree1).toMatchSnapshot();

  const component2 = renderer.create(<Container hasWon={false} boxState={['O', 'X', '', 'O', 'X', 'X', 'O', '', '']}  onIndexClick={() => {}} />);
  let tree2 = component2.toJSON();
  expect(tree2).toMatchSnapshot();

  const component3 = renderer.create(<Container hasWon={true} boxState={['O', 'X', '', 'O', 'X', 'X', 'O', '', '']}  onIndexClick={() => {}} />);
  let tree3 = component2.toJSON();
  expect(tree3).toMatchSnapshot();
});
