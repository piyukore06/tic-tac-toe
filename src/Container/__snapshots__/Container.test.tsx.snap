// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders proper hasWon and boxState output 1`] = `
<div
  className="container"
>
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  />
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  />
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  />
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  />
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  />
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  />
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  />
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  />
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  />
</div>
`;

exports[`renders proper hasWon and boxState output 2`] = `
<div
  className="container"
>
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  >
    O
  </button>
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  >
    X
  </button>
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  />
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  >
    O
  </button>
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  >
    X
  </button>
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  >
    X
  </button>
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  >
    O
  </button>
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  />
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  />
</div>
`;

exports[`renders proper hasWon and boxState output 3`] = `
<div
  className="container"
>
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  >
    O
  </button>
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  >
    X
  </button>
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  />
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  >
    O
  </button>
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  >
    X
  </button>
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  >
    X
  </button>
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  >
    O
  </button>
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  />
  <button
    className="box"
    disabled={false}
    onClick={[Function]}
  />
</div>
`;
